rotkehlchen/db/upgrades/v32_v33.py

```diff
@@ -6,7 +6,45 @@
     from rotkehlchen.db.dbhandler import DBHandler
 
 
-def _refactor_xpub_mappings(cursor: 'Cursor') -> None:
+def _refactor_xpubs_and_xpub_mappings(cursor: 'Cursor') -> None:
+    # Keep a copy of the xpub_mappings because it will get deleted once
+    # xpubs table is dropped.
+    xpub_mappings = cursor.execute('SELECT * FROM xpub_mappings').fetchall()
+    # Create a xpubs table with nullable blockchain column
+    cursor.execute("""
+    CREATE TABLE xpubs_copy (
+        xpub TEXT NOT NULL,
+        derivation_path TEXT NOT NULL,
+        label TEXT,
+        blockchain TEXT NULL,
+        PRIMARY KEY (xpub, derivation_path, blockchain)
+    );
+    """)
+    cursor.execute("""
+    INSERT INTO xpubs_copy(xpub, derivation_path, label)
+    SELECT xpub, derivation_path, label FROM xpubs;
+    """)
+    cursor.execute('UPDATE xpubs_copy SET blockchain="BTC" WHERE blockchain IS NULL;')
```

Comment:
I saw this logic here and was a bit redundant for me. Why not just avoid this creation of a table that you later recreate with the NULL restriction and in the insert do 
```suggestion
    CREATE TABLE xpubs_copy (
        xpub TEXT NOT NULL,
        derivation_path TEXT NOT NULL,
        label TEXT,
        blockchain TEXT NOT NULL,
        PRIMARY KEY (xpub, derivation_path, blockchain)
    );
    """)
    cursor.execute("""
    INSERT INTO xpubs_copy(xpub, derivation_path, label, blockchain)
    SELECT xpub, derivation_path, label, ""BTC" FROM xpubs;
    """)
    cursor.execute('UPDATE xpubs_copy SET blockchain="BTC" WHERE blockchain IS NULL;')
```

Reply:
This way you can also remove table creations, copies and renames from below

Reply:
Didn't know you could just set a default in `SELECT`. Thanks